presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergEqualityDeleteAsJoin.java

```diff
@@ -251,7 +253,11 @@ public PlanNode visitTableScan(TableScanNode node, RewriteContext<Void> context)
                     new SpecialFormExpression(SpecialFormExpression.Form.IS_NULL, BooleanType.BOOLEAN,
                             new SpecialFormExpression(SpecialFormExpression.Form.COALESCE, BigintType.BIGINT, deleteVersionColumns)));
 
-            return filter;
+            Assignments.Builder assignmentsBuilder = Assignments.builder();
+            filter.getOutputVariables().stream()
+                    .filter(variableReferenceExpression -> !variableReferenceExpression.getName().startsWith(DATA_SEQUENCE_NUMBER_COLUMN_HANDLE.getName()))
```

Comment:
nit: I think it's a little cleaner to use `!IcebergMetadataColumn.isMetadataColumnId` here

Reply:
`!IcebergMetadataColumn.isMetadataColumnId`  doesn't work here,

Reply:
Ah you only have the variable references. I guess it's fine like this 👍